src/estimators/any_sketch.py

```diff
@@ -90,6 +91,52 @@ def register_probs(self):
     return np.ones(self.num_values) / self.num_values
 
 
+class GeometricDistribution(Distribution):
+  """Distributes indexes in the range [0, num_values) according to geometric distribution."""
+
+  def __init__(self, num_values, probability=0.08):
+    """Create a truncated geometric distribution.
+    Args:
+      num_values: The number of values that the geometric distribuition can take
+        on. This is used to generate register_bounds according to geometric distribution.
+      probability: probablity of geometric distribution
+    """
+    self.num_values = num_values
+    self.probability = probability
+    self.register_bounds = GeometricDistribution._compute_register_bounds(num_values, probability)
+    self._register_probs = GeometricDistribution._compute_register_probs(num_values, probability)
+
+  def __len__(self):
+    return self.num_values
+
+  def __eq__(self, other):
+    return (isinstance(other, GeometricDistribution) and
+            self.num_values == other.num_values and
+            self.probability == other.probability)
+
+  @property
+  def register_probs(self):
+    return self._register_probs
```

Comment:
Just make the variable not have an underscore and remove this method.

Reply:
`register_probs` will be called anyway by `_estimate_cardinality_any`, I think we have to keep it